File: src/com/jetbrains/idear/actions/recognition/DebugActionRecognizer.java
package com.jetbrains.idear.actions.recognition;

import org.jetbrains.annotations.NotNull;

//only selected configuration
public class DebugActionRecognizer implements ActionRecognizer {

    @Override
    public boolean isMatching(@NotNull String sentence) {
        return sentence.contains("debug");
    }

    @Override
    public ActionCallInfo getActionInfo(@NotNull String sentence) {
        return new ActionCallInfo("Debug", null);
    }
}

File: src/main/java/com/jetbrains/idear/actions/recognition/DebugActionRecognizer.kt
package com.jetbrains.idear.actions.recognition

import com.intellij.openapi.actionSystem.DataContext

//only selected configuration
class DebugActionRecognizer : ActionRecognizer {
    override fun isMatching(sentence: String) = sentence.contains("debug")
    override fun getActionInfo(sentence: String, dataContext: DataContext) = ActionCallInfo("Debug")
}
